crates/modelardb_storage/src/optimizer/model_simple_aggregates.rs

```diff
@@ -176,281 +292,60 @@ fn try_new_aggregate_exec(
-            rewritten_aggregate_exprs.push(ModelAggregateExpr::new(ModelAggregateType::Min));
-        } else if aggregate_expr.as_any().is::<Max>() {
-            rewritten_aggregate_exprs.push(ModelAggregateExpr::new(ModelAggregateType::Max));
-        } else if aggregate_expr.as_any().is::<Sum>() {
-            rewritten_aggregate_exprs.push(ModelAggregateExpr::new(ModelAggregateType::Sum));
-        } else if aggregate_expr.as_any().is::<Avg>() {
-            rewritten_aggregate_exprs.push(ModelAggregateExpr::new(ModelAggregateType::Avg));
-        } else if let Some(aggregate_function_expr) = aggregate_expr
-            .as_any()
-            .downcast_ref::<AggregateFunctionExpr>()
-        {
-            // An aggregate query can be an AggregateFunctionExpr instead of built-in aggregate.
-            let aggregate_function_expr_name = aggregate_function_expr.fun().name();
-            match aggregate_function_expr_name {
-                "count" => rewritten_aggregate_exprs
-                    .push(ModelAggregateExpr::new(ModelAggregateType::Count)),
-                "min" => {
-                    rewritten_aggregate_exprs.push(ModelAggregateExpr::new(ModelAggregateType::Min))
-                }
-                "max" => {
-                    rewritten_aggregate_exprs.push(ModelAggregateExpr::new(ModelAggregateType::Max))
-                }
-                "sum" => {
-                    rewritten_aggregate_exprs.push(ModelAggregateExpr::new(ModelAggregateType::Sum))
-                }
-                "avg" => {
-                    rewritten_aggregate_exprs.push(ModelAggregateExpr::new(ModelAggregateType::Avg))
-                }
-                _ => {
-                    return Err(DataFusionError::NotImplemented(format!(
-                        "Aggregate function expression {} is currently not supported.",
-                        aggregate_function_expr_name
-                    )))
-                }
+) -> DataFusionResult<Vec<Arc<AggregateFunctionExpr>>> {
+    let aggregate_function_exprs = aggregate_exec.aggr_expr();
+    let mut rewritten_aggregate_exprs: Vec<Arc<AggregateFunctionExpr>> =
+        Vec::with_capacity(aggregate_function_exprs.len());
+
+    for aggregate_function_expr in aggregate_function_exprs {
+        let name_with_column = aggregate_function_expr.name();
+        let name_end = name_with_column.find('(').unwrap_or(name_with_column.len());
+
+        let model_aggregate_function_expr = match &name_with_column[..name_end] {
+            "count" => MODEL_AGGREGATE_COUNT.clone(),
+            "min" => MODEL_AGGREGATE_MIN.clone(),
+            "max" => MODEL_AGGREGATE_MAX.clone(),
+            "sum" => MODEL_AGGREGATE_SUM.clone(),
+            "avg" => MODEL_AGGREGATE_AVG.clone(),
+            name => {
+                return Err(DataFusionError::NotImplemented(format!(
+                    "Aggregate expression {name} is currently not supported.",
```

Comment:
I know this is not a new message, but should we remove "currently" which makes it sound somewhat like it soon will be supported

Reply:
Fixed.